File: tut/6mon/whales.pl
#!/usr/bin/perl -w

$count = 0;
$string = "";

while ($line = <STDIN>) {
    if ($line =~ /([0-9]+) (.*)/) {
        $curr = $2;
        $curr_count = $1;

        if ($curr eq $string) {
            $count += $curr_count;
        } else {
            if ($count > 0) {
                print "$count $string\n";
            }
            $count = $curr_count;
            $string = $curr;
        }
    }
}

if ($count > 0) {
    print "$count $string\n";
}

File: tut/6mon/whales.py
#!/usr/bin/python2.7 -u

import re
import sys

count = 0
string = ""

for line in sys.stdin:
  m = re.search("([0-9]+) (.*)", line)
  if m:
    curr = m.group(2)
    curr_count = int(m.group(1))

    if curr == string:
      count += curr_count
    else:
      if count > 0:
        print "%d %s" % (count, string)
      count = curr_count
      string = curr

if count > 0:
  print "%d %s" % (count, string)

